packages/ui/src/components/chat.tsx

```diff
@@ -0,0 +1,149 @@
+import { ChangeEvent, KeyboardEvent, ReactNode, RefObject } from 'react'
+
+import { Button, Icon, Input, Text } from '@/components'
+import { cn } from '@/utils/cn'
+
+// Root Container
+const Root: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="w-full h-full overflow-y-auto flex flex-col">{children}</div>
+}
+
+// Body
+const Body: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="flex-1 flex flex-col gap-6 mb-3 p-6">{children}</div>
+}
+
+// Footer
+const Footer: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="sticky bottom-0 p-6 pt-0 bg-background">{children}</div>
+}
+
+// Message Component
+interface MessageProps {
+  self?: boolean
+  time?: string
+  avatar?: ReactNode
+  actions?: ReactNode
+  children: ReactNode
+}
+
+const Message: React.FC<MessageProps> = ({ self, time, avatar, actions, children }) => {
+  return (
+    <div
+      className={cn('flex gap-x-3 content-center items-start', {
+        'place-content-end': self
+      })}
+    >
+      <div className="mt-0.5">{!self && avatar}</div>
+      <div
+        className={cn('flex flex-col gap-1.5', {
+          'w-[85%] items-end': self,
+          'w-full': !self
+        })}
+      >
+        <div
+          className={cn('text-foreground-3 leading-relaxed', {
+            'px-3 py-2 bg-background-8 rounded-lg': self
+          })}
+        >
+          {children}
+        </div>
+        <div className="flex gap-3 items-center justify-between mt-1">
+          <div>{actions && <div className="flex gap-1 items-center justify-start">{actions}</div>}</div>
```

Comment:
Is the extra `div` needed?
```suggestion
          {actions && <div className="flex gap-1 items-center justify-start">{actions}</div>}
```